### DISCUSSION TITLE
PostGIS proof of concept

### DISCUSSION THREAD

**Discussion by dantownsend** [piccolo-orm/piccolo#1335] [Ideas]
It would be great to have PostGIS support in Piccolo.

It would be a large chunk of work, so doing a proof of concept makes sense in the first instance, which provides basic functionality for common use cases.

There will most likely be a new file - something like `piccolo/columns/gis.py`.

**Comment by gmgreg** [comment]
I was able to get some Geography support in piccolo thanks to the issue and work by @Fingel (https://github.com/piccolo-orm/piccolo/issues/55). Would be great to see full support for PostgresGIS especially on the query side.  As it is need to use raw sql with PostgresGIS.

**Comment by dantownsend** [comment]
@gmgreg What do you think we would need for an MVP? PostGIS has so many features, it's tricky to know where to start.

**Comment by gmgreg** [comment]
Great question and I'm likely not the best person to answer but I'd think about it in use cases.  For example common queries might be:

1. Distance calculations (find X closest items to Point Y, perhaps using ST_Distance() and ST_DWithin() )
2. Area calculations (what is the area of the land bounded by... perhaps using ST_Area() and ST_Length() )
3. Spatial relationships (do these areas overlap... perhaps using ST_Intersects() ST_Contains() ST_Within() ST_Touches() )
4. Geocoding & reverse geocoding?
5. ...

You could start by supporting some use cases natively and others with a combination using SelectRaw.

Also - GeoAlchemy2 might be a good project to learn from (https://github.com/geoalchemy/geoalchemy2)

**Comment by dantownsend** [comment]
@gmgreg Thanks, that's really helpful!
